contrib/pyzfs/libzfs_core/_libzfs_core.py

```diff
@@ -0,0 +1,1335 @@
+    '''
+    Change encryption key on the specified dataset.
+
+    :param bytes fsname: the name of the dataset.
+    :param str crypt_cmd: the encryption "command" to be executed, currently
+        supported values are "new_key", "inherit", "force_new_key" and
+        "force_inherit".
+    :param props: a `dict` of encryption-related property name-value pairs;
+        only "keyformat", "keylocation" and "pbkdf2iters" are supported
+        (empty by default).
+    :type props: dict of bytes:Any
+    :param key: dataset encryption key data (empty by default).
+    :type key: bytes
+
+    :raises PropertyInvalid: if ``props`` contains invalid values.
+    :raises FilesystemNotFound: if the dataset does not exist.
+    :raises UnknownCryptCommand: if ``crypt_cmd`` is invalid.
+    :raises EncryptionKeyNotLoaded: if the encryption key is not currently
+        loaded and therefore cannot be changed.
+    '''
+    if props is None:
+        props = {}
+    if key is None:
+        key = bytes("")
+    else:
+        key = bytes(key)
+    cmd = {
+        'new_key':       _lib.DCP_CMD_NEW_KEY,
+        'inherit':       _lib.DCP_CMD_INHERIT,
+        'force_new_key': _lib.DCP_CMD_FORCE_NEW_KEY,
+        'force_inherit': _lib.DCP_CMD_FORCE_INHERIT,
+    }.get(crypt_cmd)
+    if cmd is None:
+        raise exceptions.UnknownCryptCommand(crypt_cmd)
+    nvlist = nvlist_in(props)
+    ret = _lib.lzc_change_key(fsname, cmd, nvlist, key, len(key))
+    errors.lzc_change_key_translate_error(ret, fsname)
+
+
+def lzc_load_key(fsname, noop, key):
+    '''
+    Load or verify encryption key on the specified dataset.
+
+    :param bytes fsname: the name of the dataset.
+    :param bool noop: if `True` the encryption key will only be verified,
+        not loaded.
+    :param key: dataset encryption key data.
+    :type key: bytes
+
+    :raises FilesystemNotFound: if the dataset does not exist.
+    :raises EncryptionKeyAlreadyLoaded: if the encryption key is already
+        loaded.
+    :raises EncryptionKeyInvalid: if the encryption key provided is incorrect.
+    '''
+    ret = _lib.lzc_load_key(fsname, noop, key, len(key))
+    errors.lzc_load_key_translate_error(ret, fsname, noop)
+
+
+def lzc_unload_key(fsname):
+    '''
+    Unload encryption key on the specified dataset.
+
+    :param bytes fsname: the name of the dataset.
+
+    :raises FilesystemNotFound: if the dataset does not exist.
+    :raises EncryptionKeyNotLoaded: if the encryption key is not currently
+        loaded.
+    '''
+    ret = _lib.lzc_unload_key(fsname)
+    errors.lzc_unload_key_translate_error(ret, fsname)
+
+def lzc_channel_program():
+    pass
+def lzc_channel_program_nosync():
+    pass
+def lzc_receive_one():
+    pass
+def lzc_receive_resumable():
+    pass
+def lzc_receive_with_cmdprops():
+    pass
+def lzc_receive_with_header():
+    pass
+def lzc_reopen():
+    pass
+def lzc_send_resume():
+    pass
+
+
+def lzc_sync(poolname, force=False):
```

Comment:
It seems rather cumbersome to expect a `dict` (`nvlist_t*`) from the caller so i changed this method signature.